lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1341,6 +1341,45 @@ struct FuncOpLowering : public OpConversionPattern<mlir::FuncOp> {
 };
 
 namespace {
+struct DFRemoveBlockPass
+    : public PassWrapper<DFRemoveBlockPass, OperationPass<handshake::FuncOp>> {
+  void runOnOperation() override {
+    auto funcOp = getOperation();
+    auto builder = OpBuilder(funcOp.getContext());
+    auto *termOp = funcOp.getBody().front().getTerminator();
+
+    for (auto &block : funcOp) {
```

Comment:
nit: Can you do something like `llvm::drop_begin(llvm::make_early_inc_range(funcOp), 1)`

That would let you erase blocks inside of the loop, and remove the check for `isEntryBlock`.

Reply:
Thanks for your suggestion, River! The code looks much better now.